src/main/scala/annoy4s/Annoy.scala

```diff
@@ -14,76 +14,103 @@
 
 package annoy4s
 
-import com.sun.jna._
+import java.util.UUID
+
+import annoy4s.Converters.KeyConverter
 import better.files._
+import com.sun.jna._
+
 import scala.io.Source
 
-class Annoy(
-  idToIndex: Map[Int, Int],
-  indexToId: Seq[Int],
+class Annoy[T](
+  idToIndex: Map[T, Int],
+  indexToId: Seq[T],
   annoyIndex: Pointer,
   dimension: Int
 ) {
   def close() = {
     Annoy.annoyLib.deleteIndex(annoyIndex)
   }
 
-  def query(vector: Seq[Float], maxReturnSize: Int) = {
+  def query(vector: Seq[Float], maxReturnSize: Int, searchK: Int) = {
```

Comment:
worth setting a default value of searchK: Int = -1 here? That way preserving backwards compatibility and making it easier for people who don't care about searchK to use the api.